resident/resident-service/src/main/java/io/mosip/resident/util/Utilitiy.java

```diff
@@ -37,211 +15,245 @@
-						attributes.put(value, (String) json.get(VALUE));
-					} else {
-						attributes.put(value, String.valueOf(object));
-					}
-				}
-			}
-
-			logger.info(LoggerFileConstant.APPLICATIONID.toString(), LoggerFileConstant.UIN.name(), id,
-					"Utilitiy::getMailingAttributes()::mailingAttributes::" + attributes);
-		} catch (IOException | ReflectiveOperationException e) {
-			throw new ResidentServiceCheckedException(ResidentErrorCode.RESIDENT_SYS_EXCEPTION.getErrorCode(),
-					ResidentErrorCode.RESIDENT_SYS_EXCEPTION.getErrorMessage(), e);
-		}
-		logger.debug(LoggerFileConstant.APPLICATIONID.toString(), LoggerFileConstant.UIN.name(), id,
-				"Utilitiy::getMailingAttributes()::exit");
-		return attributes;
-	}
-
-	private String getPreferredLanguage(JSONObject demographicIdentity) {
-		String preferredLang = null;
-		String preferredLangAttribute = env.getProperty("mosip.default.user-preferred-language-attribute");
-		if (!StringUtils.isEmpty(preferredLangAttribute)) {
-			Object object = demographicIdentity.get(preferredLangAttribute);
-			if(object!=null) {
-				preferredLang = String.valueOf(object);
-			}
-		}
-		return preferredLang;
-	}
-
-	private Set<String> getDataCapturedLanguages(JSONObject mapperIdentity, JSONObject demographicIdentity)
-			throws ReflectiveOperationException {
-		Set<String> dataCapturedLangauges = new HashSet<String>();
-		LinkedHashMap<String, String> jsonObject = JsonUtil.getJSONValue(mapperIdentity, MappingJsonConstants.NAME);
-		String values = jsonObject.get(VALUE);
-		for (String value : values.split(",")) {
-			Object object = demographicIdentity.get(value);
-			if (object instanceof ArrayList) {
-				JSONArray node = JsonUtil.getJSONArray(demographicIdentity, value);
-				JsonValue[] jsonValues = JsonUtil.mapJsonNodeToJavaObject(JsonValue.class, node);
-				for (JsonValue jsonValue : jsonValues) {
-					dataCapturedLangauges.add(jsonValue.getLanguage());
-				}
-			}
-		}
-		return dataCapturedLangauges;
-	}
-
-	private List<String> getDefaultTemplateLanguages() {
-		String defaultLanguages = env.getProperty("mosip.default.template-languages");
-		if (!StringUtils.isEmpty(defaultLanguages)) {
-			String[] lanaguages = defaultLanguages.split(",");
-			List<String> strList = Lists.newArrayList(lanaguages);
-			return strList;
-		}
-		return null;
-	}
-
-	public String getMappingJson() {
-		return residentRestTemplate.getForObject(configServerFileStorageURL + getRegProcessorIdentityJson, String.class);
-	}
+    private static final Logger logger = LoggerConfiguration.logConfig(Utilitiy.class);
+
+    @Autowired
+    private ResidentServiceRestClient residentServiceRestClient;
+
+    @Autowired
+    private TokenGenerator tokenGenerator;
+
+    @Value("${config.server.file.storage.uri}")
+    private String configServerFileStorageURL;
+
+    @Value("${resident.identityjson}")
+    private String getRegProcessorIdentityJson;
+
+    @Value("${mosip.primary-language}")

```

Comment:
not correct. Mosip doesnot use primary and secondary languages. It uses mandatory and optional languages, Check in bootstrap.properties where u removed mandatory and optional to add primary and secondary.
It should be other way around. Remove them and add mandatory/optional wherever applicable

Reply:
it is done.